File: NC/src/BackwardDifference.cpp
#include "../include/BackwardDifference.hpp"

BackwardDifference::BackwardDifference(function<double(double)> f, double h)
    : Differentiation(f, h) {}

double BackwardDifference::derivative(double x) const {
    return (func(x) - func(x - h)) / h;
}

File: Numerical_Computing_2/src/differentiation/backward_difference.py
from .numerical_differentiation import NumericalDifferentiation


class BackwardDifference(NumericalDifferentiation):
    def derivative(self, x: float) -> float:
        return (self.func(x) - self.func(x - self.h)) / self.h

    @property
    def method_name(self) -> str:
        return "Backward Difference"
